nixos/modules/virtualisation/neutron.nix

```diff
@@ -0,0 +1,369 @@
+
+    [nova]
+    region_name = RegionOne
+    project_domain_id = default
+    project_name = service
+    user_domain_id = default
+    password = ${cfg.novaServicePassword}
+    username = ${cfg.novaServiceUsername}
+    auth_url = http://localhost:35357
+    auth_plugin = password
+
+    [agent]
+    root_helper=/var/setuid-wrappers/sudo ${package_set}/bin/neutron-rootwrap ${rootwrapConf}
+
+    [keystone_authtoken]
+    auth_uri = http://localhost:5000
+    auth_url = http://localhost:35357
+    auth_plugin = password
+    project_name = service
+    project_domain_id = default
+    user_domain_id = default
+    username = ${cfg.serviceUsername}
+    password = ${cfg.servicePassword}
+
+    [database]
+    connection = mysql://neutron:${cfg.dbPassword}@${cfg.dbHost}/neutron
+
+    [oslo_concurrency]
+    lock_path = /var/lock/neutron
+
+    [oslo_messaging_rabbit]
+    rabbit_host = localhost
+  '';
+  utils_env = pkgs.buildEnv {
+    name = "utils";
+    paths = with pkgs; [package_set ebtables bridge-utils iproute procps conntrack_tools iputils dnsmasq coreutils  ipset
+      (lib.overrideDerivation iptables (attrs: { configureFlags = iptables.configureFlags + '' --disable-nftables''; })) ];
+  };
+  package_set = pkgs.python.buildEnv.override {
+    extraLibs = [ cfg.package ] ++ cfg.extraPackages;
+    ignoreCollisions = true;
+  };
+in {
+  options.virtualisation.neutron = {
+    package = mkOption {
+      type = types.package;
+      example = literalExample "pkgs.neutron";
+      description = ''
+        Neutron package to use.
+      '';
+    };
+
+    extraPackages = mkOption {
+      type = types.listOf types.package;
+      example = literalExample "pkgs.snabb-neutron";
+      default = [ ];
+      description = ''
+        List of extra Python packages to be installed in all
+        Neutron services. Useful for adding additional Neutron drivers.
+      '';
+    };
+
+    extraML2Config = mkOption {
+      type = types.str;
+      default = "";
+      description = ''
+      '';
+    };
+
+    enableSingleNode = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+          This option enables Neutron, also known as OpenStack Compute,
+          a cloud computing system, as a single-machine
+          installation.  That is, all of Neutron's components are
+          enabled on this machine. This is useful for evaluating and
+          experimenting with Neutron. However, for a real cloud
+          computing environment, you'll want to enable some of
+          Neutron's services on other machines.
+      '';
+    };
+
+    sharedMetadataSecret = mkOption {
+      default = "METADATA_SECRET";
+      description = "Shared secret to sign instance-id request";
+      example = "METADATA_SECRET";
+    };
+
+    publicInterface = mkOption {
+      default = "eth0";
+      description = "The pysical interface name used by the linuxbridge agent";
```

Comment:
This interface is added to the bridge created for an network which is 'bound' to an external one. I update the documentation to provide some help about this.